go/libraries/doltcore/diff/sql_diff.go

```diff
@@ -95,8 +95,13 @@ func (sds *SQLDiffSink) ProcRowWithProps(r row.Row, props pipeline.ReadableMap)
 			case DiffModifiedOld:
 				return nil
 			case DiffModifiedNew:
-				// TODO: minimize update statement to modified rows
-				stmt, err := sqlfmt.RowAsUpdateStmt(r, sds.tableName, sds.sch)
+				// Update Col Diffs to interface{} to avoid import loop
```

Comment:
Hacking the type system to get around import loops is generally a bad idea. Think a bit about what the types mean and how you want to modify their meaning. 

Here we want to provide `RowAsUpdateStmt` with more information in order to filter out portions of the SQL update statement that would have no effect. Specifically, we need a set of columns. The keys of `colDiffs` happen to be that set, but that object itself has a different purpose. A better approach here would be to construct a `set.StrSet` from the keys in `colDiffs` and pass that to `RowAsUpdateStmt`. 

Your current approach is pretty close to that as it is, but it's using the wrong types. `colDIffsInter` (does this mean "column diff intersection"?) doesn't need the values of `colDiffs`.